### PR TITLE
New checks overview in cluster details

### PR DESCRIPTION
**Pull request by fabriziosestito**
Adds the new checks overview (passing, warning, critical) count to the FE.
The component is reachable at the `clusters_new/<id>` route.

A follow up PR will include the broadcast of the execution completed event to the FE.

### CODE REVIEW

File: assets/js/components/ClusterDetails/ChecksResultOverviewNew.jsx
Comment (arbulu89) [inline assets/js/components/ClusterDetails/ChecksResultOverviewNew.jsx:14]: Super nice!
Maybe we could involve @jagabomb here once this is merged to think about UX options for loading/error/not checks available scenarios (not discarding the current implication)

File: assets/js/components/ClusterDetails/ChecksResultOverviewNew.test.jsx
Comment (arbulu89) [inline assets/js/components/ClusterDetails/ChecksResultOverviewNew.test.jsx:24]: Remove this extra new line here

File: assets/js/components/ClusterDetails/ClusterDetailsNew.jsx
Comment (arbulu89) [inline assets/js/components/ClusterDetails/ClusterDetailsNew.jsx:135]: let's go to the new settings page `/settings_new`

File: assets/js/state/lastExecutions.js
Comment (arbulu89) [inline assets/js/state/lastExecutions.js:26]: You could omit this `data: null` it already comes in the initial state

File: assets/js/state/sagas/lastExecutions.js
Comment (arbulu89) [inline assets/js/state/sagas/lastExecutions.js:19]: What is the `404` scenario for?
The moment where there is not any execution for this cluster yet?

File: assets/js/state/sagas/lastExecutions.js
Comment (arbulu89) [inline assets/js/state/sagas/lastExecutions.js:31]: This channel message from the backed is not implemented yet, right?

File: assets/js/components/ClusterDetails/ClusterDetailsNew.jsx
Comment (arbulu89) [inline assets/js/components/ClusterDetails/ClusterDetailsNew.jsx:87]: Let's put this constant action in a actions files, in the state. So we don't need to have this `dispatchUpdateLastExecution` function and we can simply do `dispatch(updateLastExecution(groupID))` or similar.
What do you think?
we started doing this in one of the previous changes

File: assets/js/state/selectors/lastExecutions.js
Comment (arbulu89) [inline assets/js/state/selectors/lastExecutions.js:2]: What would happen here if we don't have this entry? Like we don't have any execution yet for this group. Would this break something?
It would return `undefined` and most probably break something in the clusters view

**Review by arbulu89 — COMMENTED** [review]
hey @fabriziosestito ,
Nice draft! I have added some comments. Besides this, and with some additional tests, it looks really great

File: assets/js/components/ClusterDetails/ClusterDetailsNew.jsx
Comment (fabriziosestito) [reply assets/js/components/ClusterDetails/ClusterDetailsNew.jsx]: done

File: assets/js/state/selectors/lastExecutions.js
Comment (fabriziosestito) [reply assets/js/state/selectors/lastExecutions.js:2]: fixed the logic in the component

File: assets/js/components/ClusterDetails/ChecksResultOverviewNew.test.jsx
Comment (arbulu89) [inline assets/js/components/ClusterDetails/ChecksResultOverviewNew.test.jsx:12]: Simply with `loading` works. You don't need to add the `={true}` thing

File: assets/js/components/ClusterDetails/ChecksResultOverviewNew.jsx
Comment (fabriziosestito) [reply assets/js/components/ClusterDetails/ChecksResultOverviewNew.jsx]: I agree, i did some impro but it's not the best UI :D

File: assets/js/components/ClusterDetails/ChecksResultOverviewNew.test.jsx
Comment (arbulu89) [inline assets/js/components/ClusterDetails/ChecksResultOverviewNew.test.jsx:54]: If you want to be certain that the values correspond to the same header, you could maybe do something like:
```
const passing = screen.getByText('Passing');
expect(passing).toBeVisible();
expect(passing.parent().next().getByText(passing_count)).toBeVisible();
```
Or something similar

File: assets/js/state/lastExecutions.js
Comment (arbulu89) [inline assets/js/state/lastExecutions.js:36]: You can omit this `loading` here, the same in `setLastExecutionError`. They are false by default

File: assets/js/state/lastExecutions.test.js
Comment (arbulu89) [inline assets/js/state/lastExecutions.test.js:32]: extra space after `set`

File: assets/js/state/lastExecutions.test.js
Comment (arbulu89) [inline assets/js/state/lastExecutions.test.js:76]: One test scenario is missing. Most probably the most important one, `setLastExecution`. Some more `copy/paste` hehe

File: assets/js/state/sagas/lastExecutions.test.js
Comment (arbulu89) [inline assets/js/state/sagas/lastExecutions.test.js:34]: Let's name this to `execution`

File: assets/js/state/sagas/lastExecutions.test.js
Comment (arbulu89) [inline assets/js/state/sagas/lastExecutions.test.js:49]: something missing in the text. `should update the..` maybe?

**Review by arbulu89 — APPROVED** [review]
Simply great!
I have written some small updates you can apply. No need for a new review from my side

File: assets/js/state/lastExecutions.test.js
Comment (fabriziosestito) [reply assets/js/state/lastExecutions.test.js:76]: woops how did i miss this

**Review by nelsonkopliku — APPROVED** [review]
Great stuff!

The only thing that I can mention is actually related to wanda.
I noticed that if we provide to `/api/checks/groups/<group_id>/executions/last` a group id that is not in the database, the api will return an execution anyway.

**Comment by fabriziosestito** [comment]
> Great stuff!
> 
> The only thing that I can mention is actually related to wanda. I noticed that if we provide to `/api/checks/groups/<group_id>/executions/last` a group id that is not in the database, the api will return an execution anyway.

@nelsonkopliku I checked this out and in fact, this happens, the query on wanda side must be wrong. I will investigate thanks.

**Comment by rtorrero** [comment]
Great stuff @fabriziosestito, looking into how to use this for the check results overview

File: assets/js/components/ClusterDetails/ClusterDetailsNew.jsx
Comment (dottorblaster) [inline assets/js/components/ClusterDetails/ClusterDetailsNew.jsx:111]: For the future: this should be a selector

File: assets/js/components/ClusterDetails/ClusterDetailsNew.jsx
Comment (dottorblaster) [inline assets/js/components/ClusterDetails/ClusterDetailsNew.jsx:212]: Yikes!

Comment (dottorblaster) [reply assets/js/components/ClusterDetails/ClusterDetailsNew.jsx:212]: (given that `explicit is better than implicit` your call if you want to just splat this or set prop by prop)

Comment (dottorblaster) [reply assets/js/components/ClusterDetails/ClusterDetailsNew.jsx:212]: > `explicit is better than implicit`

Follow me for more proverbs

File: assets/js/state/actions/lastExecutions.js
Comment (dottorblaster) [inline assets/js/state/actions/lastExecutions.js:1]:
```suggestion
export const UPDATE_LAST_EXECUTION = 'UPDATE_LAST_EXECUTION';
```

File: assets/js/state/actions/lastExecutions.js
Comment (dottorblaster) [inline assets/js/state/actions/lastExecutions.js:5]:
```suggestion
    type: UPDATE_LAST_EXECUTION,
```

File: assets/js/state/sagas/lastExecutions.js
Comment (dottorblaster) [inline assets/js/state/sagas/lastExecutions.js:3]:
```suggestion
import { UPDATE_LAST_EXECUTION } from '@state/actions/lastExecutions';
```

File: assets/js/state/sagas/lastExecutions.js
Comment (dottorblaster) [inline assets/js/state/sagas/lastExecutions.js:35]:
```suggestion
  yield takeEvery(UPDATE_LAST_EXECUTION, updateLastExecution);
```

**Review by dottorblaster — CHANGES_REQUESTED** [review]
Just a tiny change that has a bit of impact, but really good job!

**Review by dottorblaster — APPROVED** [review]
LGTM

**Pull request merged**
